src/controllers/course.ts

```diff
@@ -1,87 +1,75 @@
-import { Request, Response} from 'express';
-import {AppDataSource} from "../config/data-source"
-import {Course} from "../entity/course" 
+import { Request, Response } from 'express';
+import Course from "../model/course";
 
-class CourseController{
-    public static async index(req: Request, res: Response){
-        const course = await AppDataSource.manager.find(Course)
-        return res.status(200).json({
-            message: course
-        });  
+class CourseController {
+    public static async index(req: Request, res: Response) {
+        try {
+            const course = await Course.findAll();
+            return res.status(200).json(course);
+        } catch (error: any) {
+            console.error('An error occurred:', error.message); // Log the error message
```

Comment:
Create a reusable response generator that can send different responses for specific error codes.  We can have multiple different types of error and for each error we may need to return a different response so user knows what's going on. Also there should be an error code returned  along with the message.

Reply:
Done!